system/modules/core/pages/PageRegular.php

```diff
@@ -87,6 +89,7 @@ public function generate($objPage, $blnCheckRequest=false)
 			{
 				while ($objModules->next())
```

Comment:
Jup. I stopped using `->next()` because it can lead to issues (due to the internal position pointer). As an example, if you use `getRelated` with a collection as result, each call would get you the same collection with the previous internal pointer (possibly set to the last result, so `next()` would return `false`).